node/core/approval-voting/src/lib.rs

```diff
@@ -1880,16 +1910,58 @@ fn process_wakeup(
 				"Launching approval work.",
 			);
 
-			// sanity: should always be present.
-			actions.push(Action::LaunchApproval {
-				indirect_cert,
-				assignment_tranche: tranche,
-				relay_block_number: block_entry.block_number(),
-				candidate_index: i as _,
-				session: block_entry.session(),
-				candidate: candidate_entry.candidate_receipt().clone(),
-				backing_group,
-			});
+			let mut approvals = if let Ok(app) = state.approvals.lock() {
+				app
+			} else {
+				return Ok(Vec::new());
```

Comment:
If we experience a poisoned mutex, we won't recover from it. Would it make more sense to simply unwrap the lock?

Or should we return an Error instead? I was hesitant to add a SubsystemError variant for poisoned Mutex as I'm not super happy with having needed a Mutex in the first place.

Reply:
Addressed by architectural comment above.